src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -262,345 +280,488 @@ public enum SupremeVictors implements LogicCardInfo {
           weakness W
           move "Luring Flame", {
-            text "Switch the Defending Pokémon with 1 of your opponent’s Benched Pokémon. The new Defending Pokémon is now Burned."
+            text "Switch the Defending Pokémon with 1 of your opponent's Benched Pokémon. The new Defending Pokémon is now Burned."
             energyCost R
-            attackRequirement {}
+            attackRequirement {
+              assert opp.bench : "Your opponent has no Benched Pokémon"
+            }
             onAttack {
-              damage 0
+              def target = opp.bench.select("Switch 1 of your opponent’s Benched Pokémon with the Defending Pokémon.")
+              if ( sw2(target) ) { apply BURNED, target }
             }
           }
           move "Vapor Kick", {
-            text "30+ damage. Pokémon in play, this attack does 30 damage plus 30 more damage."
-            energyCost R, C, W
+            text "30+ damage. If your opponent has any [W] Pokémon in play, this attack does 30 damage plus 30 more damage."
+            energyCost R, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              if(opp.all.find{it.types.contains(W)}) {
+                damage 30
+              }
             }
           }
-
         };
       case DRIFBLIM_FB_3:
-        return basic (this, hp:HP080, type:PSYCHIC, retreatCost:2) {
+        return basic (this, hp:HP080, type:P, retreatCost:2) {
           weakness D
           resistance C, MINUS20
           pokeBody "Pump Up", {
-            text "If your opponent has 3 or less Prize cards left, Drifblim gets +40 HP."
+            text "If your opponent has 3 or less Prize cards left, Drifblim FB gets +40 HP."
             delayedA {
+              getterA GET_FULL_HP, self, {h->
+                if(self.owner.opposite.pbg.prizeCardSet.size() <= 3) {
+                  h.object += hp(40)
+                }
+              }
             }
           }
           move "Shadow Ball", {
-            text "Choose 1 of your opponent’s Benched Pokémon. This attack does 40 damage to that Pokémon. Apply Weakness and Resistance for this attack."
+            text "Choose 1 of your opponent's Benched Pokémon. This attack does 40 damage to that Pokémon. Apply Weakness and Resistance."
             energyCost P, C, C
-            attackRequirement {}
+            attackRequirement {
+              assert opp.bench : "Your opponent has no Benched Pokémon"
+            }
             onAttack {
-              damage 0
+              def pcs = opp.all.select()
+              new ResolvedDamage(hp(40), self, pcs, ATTACK, DamageManager.DamageFlag.FORCE_WEAKNESS_RESISTANCE).run(bg)
+              bg.dm().applyWeakness()
+              bg.dm().applyResistance()
+              def damage = bg.dm().getTotalDamage(self, pcs)
+              bg.dm().clearDamages()
+              bg.em().run(new DirectDamage(damage, pcs))
```

Comment:
Why is this so much more complicated than when Galvantula STS applies Weak/Resist to the bench? It looks like it may be partially based on Elekid N1? However, that is only because Elekid's is not an attack.

Reply:
This was taken directly from galvantula UNM 62. The STS code does look much better than this.